hoodie-client/src/main/java/com/uber/hoodie/index/HoodieIndex.java

```diff
@@ -115,6 +115,10 @@ protected HoodieIndex(HoodieWriteConfig config) {
    */
   public abstract boolean isImplicitWithStorage();
 
+  /**
+   * Each index type should implement it's own logic to release any resources acquired during the process.
+   */
+  public void close() {}
```

Comment:
Having an implementation of close() for an Index does not sound very intuitive. I think we should remove this method and then add a shutdown hook in HbaseIndex only and add anything related to releasing resources or even a private method that does it after updating all the indexes. Also, I don't see this method being called anywhere.

Reply:
The only reason to add explicit close() method as opposed to adding it as part of shutdownhook or AutoCloseable is to be able to release resources as quickly as possible for other shared ingestion pipelines. This will allow the index to release any resources acquired for dynamic QPS implementation, right after HBase puts. 
Intended to call this from HoodieWriteClient. Good catch, updating now.